unblob/handlers/archive/ewf/ewf.py

```diff
@@ -0,0 +1,202 @@
+    def __init__(self, header_struct: str):
+        self.header_struct = header_struct
+        self._struct_parser = StructParser(C_DEFINITIONS)
+
+    def table_descriptor(
+        self, file: File, position: int, outdir: Path, sectors_per_chunk: int
+    ) -> Instance:
+        fs = FileSystem(outdir)
+        entries = []
+        header = self._struct_parser.parse("table_descriptor_t", file, Endian.LITTLE)
+        entry_path = Path("ewf.decrypted")
+
+        for _ in range(header.number_of_entries):
+            entry = self._struct_parser.parse("table_entry_t", file, Endian.LITTLE)
+            entries.append(entry.offset)
+
+        with fs.open(entry_path) as output_file:
+            for offset in entries:
+                file.seek(
+                    position
+                    + int.from_bytes(offset, byteorder="little")
+                    - DESCRIPTOR_LEN,
+                    io.SEEK_SET,
+                )
+
+                magic_bytes = file.read(2)
+                compressed = any(magic_bytes == magic.value for magic in ZlibMagic)
+
+                for chunk in iterate_file(
+                    file,
+                    position
+                    + int.from_bytes(offset, byteorder="little")
+                    - DESCRIPTOR_LEN,
+                    sectors_per_chunk,
+                ):
+                    if compressed:
+                        compressed_chunk = zlib.decompress(chunk)
+                        output_file.write(compressed_chunk)
+                    output_file.write(chunk)
+
+    def extract(self, inpath: Path, outdir: Path):
+        with File.from_path(inpath) as file:
+            file.seek(EWF_HEADER_LEN)  # we skip the initial header
+            data_descriptor = self._struct_parser.parse(
+                "data_descriptor_t", file, Endian.LITTLE
+            )
+            logger.debug("data_descriptor_t", header=data_descriptor, _verbosity=3)
+
+            # the file is made of section, we loop over all the sections
+            while data_descriptor.definition != Definition.DONE.value:
+                file.seek(data_descriptor.next_offset, io.SEEK_SET)
+                data_descriptor = self._struct_parser.parse(
+                    "data_descriptor_t", file, Endian.LITTLE
+                )
+                logger.debug("data_descriptor_t", header=data_descriptor, _verbosity=3)
+
+                if data_descriptor.definition == Definition.VOLUME.value:
+                    volume_descriptor = self._struct_parser.parse(
+                        "volume_descriptor_t", file, Endian.LITTLE
+                    )
+                    sectors_per_chunk = find_chunk_size(volume_descriptor)
+
+                if data_descriptor.definition == Definition.SECTORS.value:
+                    position = file.tell()
+
+                if data_descriptor.definition == Definition.TABLE.value:
+                    self.table_descriptor(file, position, outdir, sectors_per_chunk)
```

Comment:
Can you explain how EWF is structured ? Can we observe VOLUME, SECTORS, and TABLE in random orders ? Do they follow a strict ordering ? Is it possible to have multiple VOLUMEs ? What about the others ?